makefiles/app_dirs.inc.mk

```diff
@@ -3,16 +3,55 @@
 # fallback so empty RIOTBASE won't lead to "/examples/"
 RIOTBASE ?= .
 
+# Define the list of examples subdirectories that contain application directories
+EXAMPLES_APPLICATIONS_SUBDIRS :=  \
+    ble                       \
+    coap                      \
+    cord                      \
+    dtls                      \
+    gnrc                      \
```

Comment:
I think this could change with the discussion below, but I'm not sure if adding `gnrc` here instead of `ipv6`, `udp` is a good idea. I expect that most newcomers would try to search for those and `gnrc` doesn't say too much unless someone uses RIOT beforehand.

For instance, where would we put GNRC LoRaWAN? `gnrc`? `lora`?

Reply:
Sorry, I just saw https://github.com/RIOT-OS/RIOT/pull/15358#issuecomment-777772212.

But in any case, I still think it would be better to have a more "functional" name than `gnrc` here. All the examples mentioned in https://github.com/RIOT-OS/RIOT/pull/15358#issuecomment-777772212 have different categories (`gnrc_border_router` is a border router, gnrc_networking_mac is just using one MAC layer, etc).

I think it would make sense something like:
```
- border_routers
    - gnrc_border_router
    - openthread_ncp (coming soon)
- ieee802154
    - lwmac
    - gomac
    - submac
-  netif
    - gnrc_networking
```

or something like that

Reply:
Keep in mind that one day some applications in the tests might arrive in the `examples` directory. The current proposal is a way to organize the "storage" of applications in the repo. How well it's organized is not so important IMHO and it will be difficult to find an organization that fits with the diversity of applications. In the long term, what would be beneficial is to add another layer of documentation (using Sphinx, who knows) where example applications could be referenced in different categories (networking, lorawan, gnrc => gnrc_lorawan, etc) but the application would remain stored in the `examples/lorawan`, because it's the most obvious category.

I fear that it will be impossible to find an organization that fits everyone vision and that this discussion will be endless.

Reply:
ok, I see your point and agree. I also agree that this wouldn't replace documentation

Reply:
> I fear that it will be impossible to find an organization that fits everyone vision and that this discussion will be endless.

I agree here, a first step is having an organization, moving that organization around should be easier once the initial change is in.